src/contexts/GroupsContextDispatch.jsx

```diff
@@ -0,0 +1,19 @@
+import { createContext } from 'react';
+
+const GroupsDispatchContext = createContext(null);
+
+function groupReducer(state, action) {
+  switch (action.type) {
+    case 'setIndex':
+      return { groups: state.groups, index: action.payload };
+    case 'update':
+      return { groups: action.payload, index: state.index };
+    case 'init':
+      return { groups: action.payload, index: 0 };
+    default:
+      throw new Error('Invalid Group reducer action.');
+  }
+}
+
+export default GroupsDispatchContext;
+export { groupReducer };
```

Comment:
Another nitpick but this could be simplified to
```js
export { GroupsDispatchContext as default, groupReducer }
```

Reply:
Ill make that change.